src/query/automaton_weight.rs

```diff
@@ -83,6 +77,25 @@ where
     }
 }
 
+fn automaton_score<A>(automaton: &A, state: A::State) -> f32
+where
+    A: Automaton + Send + Sync + 'static,
+    A::State: Clone,
+{
+    if TypeId::of::<DFAWrapper>() == automaton.type_id() && TypeId::of::<u32>() == state.type_id() {
+        let dfa = automaton as *const A as *const DFAWrapper;
+        let dfa = unsafe { &*dfa };
+
+        let id = &state as *const A::State as *const u32;
+        let id = unsafe { *id };
+
+        let dist = dfa.0.distance(id).to_u8() as f32;
+        1.0 / (1.0 + dist)
```

Comment:
Can we document this score somewhere in the doc of the FuzzyTermScorer?

Ideally I think this should be implemented as an array `[f32; 5]` that is passed by the user. (5 is sufficient. We do not handle any levenshtein distance above 4.)

I'm ok with `[1f32, 0.5f32, 0.3333f32, ...]` as a default.

Reply:
Just to be sure, isn't max distance levenshtein 2 ? In this [range](https://github.com/tantivy-search/tantivy/blob/main/src/query/fuzzy_query.rs#L38)

Reply:
Or the transposition below multiplies it by 2 again ?

Reply:
Ah yes you are right!
The limit of 4 is within the levenshtein automata library. I reduced it in tantivy.